docs/migrations/v4-to-v5.md

```diff
@@ -0,0 +1,390 @@
+# Migrating from v4 to v5
+
+This document is intended to highlight significant changes which may require more information than presented in the CHANGELOG.
+Any changes that must be done by a user of ibc-go should be documented here.
+
+There are four sections based on the four potential user groups of this document:
+- Chains
+- IBC Apps
+- Relayers
+- IBC Light Clients
```

Comment:
Food for thought, going forward we could add directories for each migration v4-to-v5 and within each directory is a file for each user. So the relayer only needs to open the relayer file and realize no changes should be necessary with respect to ibc-go

Reply:
It might be nice on the [docs website](https://ibc.cosmos.network/main/migrations/v3-to-v4.html), if there was subsections that could be clicked on for each interested party

Reply:
I like @chatton's idea of having these as links. I just pushed an update.
We can iterate later on if we should move each to a separate file.